homeassistant/components/roborock/__init__.py

```diff
@@ -32,39 +31,58 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     _LOGGER.debug("Getting home data")
     home_data = await api_client.get_home_data(user_data)
     _LOGGER.debug("Got home data %s", home_data)
-    devices: list[HomeDataDevice] = home_data.devices + home_data.received_devices
+    device_map: dict[str, HomeDataDevice] = {
+        device.duid: device for device in home_data.devices + home_data.received_devices
+    }
+    product_info = {product.id: product for product in home_data.products}
     # Create a mqtt_client, which is needed to get the networking information of the device for local connection and in the future, get the map.
-    mqtt_client = RoborockMqttClient(
-        user_data, {device.duid: RoborockDeviceInfo(device) for device in devices}
-    )
+    mqtt_clients = [
+        RoborockMqttClient(
+            user_data, DeviceData(device, product_info[device.product_id].model)
+        )
+        for device in device_map.values()
+    ]
     network_results = await asyncio.gather(
-        *(mqtt_client.get_networking(device.duid) for device in devices)
+        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients)
     )
     network_info = {
         device.duid: result
-        for device, result in zip(devices, network_results)
+        for device, result in zip(device_map.values(), network_results)
         if result is not None
     }
-    try:
-        await mqtt_client.async_disconnect()
-    except RoborockException as err:
-        _LOGGER.warning("Failed disconnecting from the mqtt server %s", err)
+    await asyncio.gather(
```

Comment:
Should this still catch `RoborockException` here and below or is that now handled by the empty map check?

Aside: While not required, these types of failure cases may be tricky enough that you'd want to add tests for them if they aren't there already.

Reply:
It's handled by the empty map check, as disconnects don't matter if they fail, but I should probably add a warning to let the user know(even though there isn't much they can do)

However, I'm on vacation and won't be back until Wednesday, so I wouldn't be able to test it, so I'm not sure if it is worth adding a log for now. But your call, I can do it when I get back.

Reply:
Got it, `eturn_exceptions=True`. OK, yes, disconnects don't matter here.